### DISCUSSION TITLE
Removing params from the url

### DISCUSSION THREAD

**Discussion by monks1975** [paoloricciuti/sveltekit-search-params#27] [Q&A]
Is it currently possible to remove a param completely from the url after it has been added when using:

```js
<script lang="ts">
    import { queryParameters } from "sveltekit-search-params";

    const store = queryParameters();
</script>
```

I found it is possible when making individual stores, but that doesn't fit my use case.

I have a multiselect input component and can add params from it (which show as tags) like ?filter=smith to filter server data. Removing the param then would reload with non-filtered data.

**Comment by tordans** [comment]
> I found it is possible when making individual stores, 

@monks1975 how did you do it when using separate stores?

I have `const urlSearch = queryParam('q')` and want to remove the `?q=` every time the search value is `null`/`undefined`.
